test/functional/ui/float_spec.lua

```diff
@@ -2901,6 +2901,56 @@ describe('float window', function()
         end
       end)
 
+      it('builtin popupmenu rendered above cursor (pum_above)', function()
+        feed('i<cr><cr>x ')
+        funcs.complete(3, {'aa', 'word', 'longtext'})
+        if multigrid then
+          screen:expect{grid=[[
+          ## grid 1
+            [2:----------------------------------------]|
+            [2:----------------------------------------]|
+            [2:----------------------------------------]|
+            [2:----------------------------------------]|
+            [2:----------------------------------------]|
+            [2:----------------------------------------]|
+            [3:----------------------------------------]|
+          ## grid 2
+                                                    |
+            {0:~                                       }|
+            {0:~                                       }|
+            {0:~                                       }|
+            {0:~                                       }|
+            {0:~                                       }|
+          ## grid 3
+            {3:-- INSERT --}                            |
+          ## grid 4
+            {7:            }|
+            {7:            }|
+            {7:x aa^        }|
+            {12:~           }|
+          ## grid 5
+            {13: aa             }|
+            {1: word           }|
+            {1: longtext       }|
+          ]], float_pos={
+            [4] = {{id = 1001}, "NW", 1, 2, 5, true, 50 },
+            -- next line is still incorrect, i expect SW to be there,
+            -- but actual result is NW, and grid doesn't look right
+            [5] = {{id = -1}, "SW", 4, 3, 1, false, 100 },
```

Comment:
@bfredl please take a look at this.

`screen:expect` without multigrid looking fine, but with multigrid i exepct here to be SW, but actual result is NW.

any ideas what i doing wrong here?

Reply:
Is it wrong tho? It just seems it decided to place the popupmenu below. I'd suggest using the cmdline as in your original screenshot, to add a test which takes the SW branch.

Reply:
That's nice idea.

Please take a look, i updated the test and it does what i expect.